stock_picking_batch_extended/wizard/stock_picking_to_batch.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2012-2016 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import json
+
+from odoo import api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockPickingToBatch(models.TransientModel):
+    """Create a stock.picking.batch from stock.picking"""
+
+    _inherit = "stock.picking.to.batch"
+    _group_field_param = "stock_batch_picking.group_field"
+
+    name = fields.Char(
+        help="Name of the batch picking",
+    )
+    user_id = fields.Many2one(
+        default=lambda self: self._default_user_id(),
+    )
+    notes = fields.Text(help="Free form remarks")
+    batch_by_group = fields.Boolean(
+        string="Create batch pickings grouped by fields",
+    )
+    group_field_ids = fields.One2many(
+        comodel_name="stock.picking.batch.creator.group.field",
+        inverse_name="picking_to_batch_id",
+        string="Group by field",
+        help="If set any, multiple batch picking will be created, one per "
+        "group field",
+    )
+
+    @api.onchange("batch_by_group")
+    def onchange_batch_by_group(self):
+        if self.batch_by_group:
+            self.group_field_ids = False
+            for index, field in enumerate(self.load_store_fields()):
+                self.group_field_ids += self.group_field_ids.new(
+                    {"sequence": index, "field_id": field.id}
+                )
+
+    def load_store_fields(self):
+        group_field_ids = (
+            self.env["ir.config_parameter"].sudo().get_param(self._group_field_param)
+        )
+        group_fields = self.env["ir.model.fields"].browse(
+            group_field_ids and json.loads(group_field_ids)
+        )
+        return group_fields
+
+    @api.model
+    def default_get(self, fields):
+        """
+        Set last grouped fields used that are stored in config parameters
+        """
+        res = super().default_get(fields)
+        group_fields = self.load_store_fields()
+        res["batch_by_group"] = group_fields and True or False
+        return res
+
+    def _default_user_id(self):
+        """Return default_user_id from the main company warehouse
+        except if a warehouse_id is specified in context.
+        """
+        warehouse_id = self.env.context.get("warehouse_id")
+        if warehouse_id:
+            warehouse = self.env["stock.warehouse"].browse(warehouse_id)
+        else:
+            warehouse = self.env["stock.warehouse"].search(
+                [("company_id", "=", self.env.user.company_id.id)], limit=1
+            )
+
+        return warehouse.default_user_id
+
+    def _prepare_stock_batch_picking(self):
+        vals = {
+            "notes": self.notes,
```

Comment:
```suggestion
            "notes": self.notes,
            "description": self.description,
```

Description field is missed:
https://github.com/odoo/odoo/blob/21993c1f8edca2ed7f40d68095f4b32e767525de/addons/stock_picking_batch/models/stock_picking_batch.py#L20

Perhaps we must use attach_pickings with default_name  value in context...